containers/java/.devcontainer/library-scripts/gradle-debian.sh

```diff
@@ -128,6 +128,6 @@ fi
 
 # Install gradle
 sdk_install gradle ${GRADLE_VERSION} '\s\s' '\s\s' '^[0-9]+\.[0-9]+\.[0-9]+$'
-updaterc '[ -z "${GRADLE_USER_HOME}" ] && export GRADLE_USER_HOME=${HOME}/.gradle'
+updaterc '[ -z "${GRADLE_USER_HOME}" ] && export GRADLE_USER_HOME=$HOME/.gradle'
```

Comment:
What's the difference between ${HOME} vs $HOME here?

Reply:
Look at the maven script, it doesn't have the curlies. I believe the script is running in a mode where some variables are not defined. Having the curlies makes the variable being interpreted to nothing, removing them make the variable a literal, which is interpreted only when the newly defined environnement variable is called, at a time home is defined.

Reply:
ping @testforstephen

Reply:
Sorry for the late response.

According to https://www.gnu.org/software/bash/manual/html_node/Single-Quotes.html, since we use single quote to wrap the whole text, it preserves the literal value when passing it as a parameter to the function `updaterc`. It looks no difference between using `${HOME}` and `$HOME` here.

Look at what the function `updaterc` does, it just appends the argument to the `bash.bashrc` and `zshrc` files.
https://github.com/microsoft/vscode-dev-containers/blob/e3681de503e0d8acf05ce6e0cc5fc92da4d22393/containers/java/.devcontainer/library-scripts/gradle-debian.sh#L46-L55

And I also checked my remote Java container, here is the value of running the shell `cat /etc/bash.bashrc`. It exactly preserves the literal value.

![image](https://user-images.githubusercontent.com/14052197/192683331-d4e5aa57-8964-4d9f-933c-d7b987b2bc91.png)

@baywet Could you check what the command `cat /etc/bash.bashrc` returns in your environment?